src/java.base/share/classes/java/lang/invoke/SimpleMethodHandle.java

```diff
@@ -32,6 +32,8 @@
 
 /**
  * A method handle whose behavior is determined only by its LambdaForm.
+ * Access to SimpleMethodHandle should ensure BoundMethodHandle is initialized
+ * first.
```

Comment:
Did you try factoring UNSAFE.ensureInit(BMH) into a static block in SimpleMethodHandle.java?
Sometimes that works.

Reply:
To be exact, I did not encounter any initialization issue during my use of BMH/SimpleMH. We don't need ensureInit as SimpleMH already extends BMH. Maybe my concern is just red herring, as I saw a warning on BMH$Species_L that its access should go through BMH first.